osu.Game/Beatmaps/Drawables/BeatmapSetOnlineStatusPill.cs

```diff
@@ -99,12 +98,20 @@ protected override void LoadComplete()
 
         private void updateState()
         {
-            if (Status == BeatmapOnlineStatus.None)
+            if (Status == BeatmapOnlineStatus.None && !ShowUnknownStatus)
             {
                 Hide();
                 return;
             }
 
+            // Only animate resizing if we already have a size.
+            // This avoids animating height from zero.
+            if (Width > 0)
```

Comment:
Why does the comment say something about height but actually check width?

Reply:
I guess it can use height without an issue. Either works in this scenario, width just made more sense in my head since the animation is here specifically for animating width changes.